### DISCUSSION TITLE
umi 3x 能切换成pnpm吗

### DISCUSSION THREAD

**Discussion by simplast** [umijs/umi#12460] [General]
1

**Comment by fz6m** [comment]
umi 3 不维护了，建议升级到 umi 4 。

对于现在社区内不能使用 pnpm 的项目，现在较好的方法是开启 [`shamefully-hoist=true`](https://pnpm.io/npmrc#shamefully-hoist) 变成打平的依赖结构，强行使用。

或者使用 yarn v4 版本，速度也和 pnpm 一样快，甚至有时比 pnpm 更快。
